zerberus/voice.cpp

```diff
@@ -160,8 +162,24 @@ void Voice::start(Channel* c, int key, int v, const Zone* z)
       envelopes[V1Envelopes::DECAY].setTime(z->ampegDecay, _zerberus->sampleRate());
       envelopes[V1Envelopes::DECAY].offset = z->ampegSustain;
 
+      // The duration of the sustain stage depends on the trigger event
+      //    if it is triggered by note on it is virtually infinite (until there is the note off event)
+      //    hence the voice is also entering the "play" state
+      //
+      //    if it is triggered by noteoff (release) there is no signal that is going to stop the voice
+      //    hence we let it play as long as the sample file duration
+      //    to get that we take the duration of the file and subtract the durations of the other
+      //    envelope stages
+
       envelopes[V1Envelopes::SUSTAIN].setTable(Envelope::egLin);
-      envelopes[V1Envelopes::SUSTAIN].setTime(std::numeric_limits<float>::infinity(), _zerberus->sampleRate());
+      if (trigger == Trigger::RELEASE) {
+            double sampleDur = (z->sample->frames()/z->sample->channel()) / z->sample->sampleRate();
```

Comment:
Sample is played on noteoff. We need to stop the voice when it's done. Set the sustain duration accordingly.